.github/workflows/main.yml

```diff
@@ -197,7 +194,6 @@ jobs:
   test_client_js_legacy:
     name: Test Client JS Legacy
     runs-on: ubuntu-latest
-    needs: format_and_lint_client_js_legacy
```

Comment:
I think the `test_client_js_legacy` needs to depend on `build_programs` since it needs the binary of the program, although I could not see the "restore" step here.

Reply:
As mentioned earlier, the JS tests don't actually run against a validator or a program, they're just unit tests